redfish/collector.go

```diff
@@ -0,0 +1,241 @@
+package redfish
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/cybozu-go/log"
+	"github.com/cybozu-go/setup-hw/config"
+	"github.com/cybozu-go/setup-hw/gabs"
+	"github.com/prometheus/client_golang/prometheus"
+	yaml "gopkg.in/yaml.v2"
+)
+
+const namespace = "hw"
+
+// Collector implements prometheus.Collector interface.
+type Collector struct {
+	rules  []convertRule
+	client *client
+	cache  *cache
+}
+
+// CollectorConfig is a set of configurations for Collector.
+type CollectorConfig struct {
+	AddressConfig *config.AddressConfig
+	Port          string
+	UserConfig    *config.UserConfig
+	Rule          []byte
```

Comment:
I think `[]byte` should not be used here.

`convertRule` type should be exported, and this should be `[]ConvertRule`.